### ISSUE TITLE
failed real servers stay out of VS

### ISSUE THREAD

**Issue by markrechler** [acassen/keepalived#362]
Running `keepalived-1.2.13-7.el7.x86_64` it seems that failed real servers never get checked again after being removed. Sending HUP to the process seems to be the only way to get the servers re-added/checked. Is this expected behavior? Using `HTTP_GET` checks.

```
Timeout connect, timeout server [x.x.x.x]:80.
Removing service [x.x.x.x]:80 from VS [x.x.x.x]:80
```

**Comment by acassen** [comment]
please consider upgrading to latest 1.2.22... bunch of fixes and extensions have been made here...

**Comment by jehiah** [comment]
@acassen FYI: it wasn't for lack of trying a newer version. #363 is from our attempt at using 1.2.22 first. 

It would be helpful in our troubleshooting this issue (real servers being removed from rotation permanently) to have some confirmation that this isn't expected behavior.

**Comment by fhuitfeldt** [comment]
I was getting the described behaviour with 1.2.13 as well.

It seems that turning on 'inhibit_on_failure' produces the desired state where servers are dynamically added and removed from the pool.

**Comment by jehiah** [comment]
@fhuitfeldt this was solved in #373 as a bug when `status_code` criteria was not defined.

**Comment by fhuitfeldt** [comment]
@jehiah thank you very much for your heads up. While we are certainly able to build keepalived ourselves, we are very conservative at my company, so we are fairly tightly bound to "official" packages, which currently leaves us at 1.2.13 (on centos & RHEL).

as a side note, just for other devs finding this thread, we *do* define status_code in our health check, but setting 'inhibit_on_failure' was still necessary as a workaround.

**Comment by pqarmitage** [comment]
@fhuitfeldt one could say the conservative/risk averse approach is to use the latest version of keepalived with the many bug fixes in the 2 1/4 years since v1.2.13 was released, and the adventurous/risky approach is to remain with version 1.2.13 :grinning:.

**Comment by fhuitfeldt** [comment]
hehe, I know and completely agree :-) There are good reasons for our conservatism, but we also continuously reevaluate.
